jwst/pipeline/calwebb_spec3.py

```diff
@@ -380,3 +422,176 @@ def _create_nrsmos_source_id(self, source_models):
+        )
+        fltdata_by_exposure.append(flt_empty)
+
+    # Now loop through the models and populate the tables
+    # Need to index each exposure separately because they may have a different number of sources
+    loop_index_by_exposure = [0] * n_exposures
+    for model in results_list:
+        # inner loop over exposures
+        for spec in model.spec:
+            # ensure data goes to table corresponding to correct exposure based on filename
+            exp_num = spec.meta.observation.exposure_number
+            exposure_idx = exposure_numbers.index(exp_num)
+            fltdata = fltdata_by_exposure[exposure_idx]
+            n_rows = n_rows_by_exposure[exposure_idx]
+            j = loop_index_by_exposure[exposure_idx]
+            loop_index_by_exposure[exposure_idx] += 1
+
+            # rename some spec metadata to clarify where they came from
+            # at the request of the WFSS teams
+            name_roots = ["XSTART", "YSTART", "XSTOP", "YSTOP"]
+            new_names = ["EXTRACT1D_" + name for name in name_roots]
+            old_names = ["EXTRACTION_" + name for name in name_roots]
+            for old_name, new_name in zip(old_names, new_names, strict=True):
+                setattr(spec, new_name.lower(), getattr(spec, old_name.lower(), None))
+
+            # populate the table with data from the input spectrum
+            populate_recarray(
+                fltdata[j], spec, n_rows, all_columns, is_vector, ignore_columns=["NELEMENTS"]
+            )
+
+            # special handling for NELEMENTS because not defined in specmeta schema
+            fltdata[j]["NELEMENTS"] = spec.spec_table.shape[0]
+
+    # Finally, create a new MultiExposureModel to hold the combined data
+    # with one MultiSpecModel table per exposure
+    output_x1d = dm.WFSSMultiExposureSpecModel()
+    for i, exposure_number in enumerate(exposure_numbers):
+        # Create a new extension for each exposure
+        spec_table = fltdata_by_exposure[i]
+        spec_table.sort(order="SOURCE_ID")
+        ext = dm.WFSSMultiSpecModel(spec_table)
+
+        # copy units from any of the SpecModels (they should all be the same)
+        copy_column_units(spec, ext)
+
+        # copy metadata
+        fname = exposure_counter[exposure_number]["filename"]
+        # ext.meta.filename = fname
+        # ext.meta.observation.exposure_number = exposure_number
+
+        output_x1d.exposures.append(ext)
+        output_x1d.exposures[-1].filename = fname
+        output_x1d.exposures[-1].exposure_number = int(exposure_number)
+
+    # Save the combined results to a file using first input model for metadata
+    example_model = results_list[0]
+    output_x1d.update(example_model, only="PRIMARY")
+    output_x1d.save(output_filename)
+
+
+def _save_wfss_c1d(results_list, output_filename):
+    """
+    Compile exposure-averaged sources into a single table and save to a file.
+
+    The output c1d product will have just one science extension.
```

Comment:
"science" is a little unclear here - you mean a table extension?

```suggestion
    The output c1d product will have just one science extension.
```